packages/react/src/components/text-input/text-input.tsx

```diff
@@ -105,12 +105,12 @@ export const TextInput = forwardRef(({
     }, [id, ariaDescribedBy, validity, hint]);
 
     const handleBlur: (event: FocusEvent<HTMLInputElement>) => void = useCallback((event) => {
-        setValidity({ validity: event.currentTarget.checkValidity() });
+        setValidity({ validity: (valid ?? true) && event.currentTarget.checkValidity() });
```

Comment:
L'idée est que si `valid` existe, on l'utilise à la place de `event.currentTarget.checkValidity()` et `validity` serait égale à `false` si `valid` est égale à `false`.
```suggestion
        setValidity({ validity: valid ?? event.currentTarget.checkValidity() });
```

Reply:
J'ai fait cela pour garder la validation native HTML 5 comme required et pattern même si valid est true.
Donc la question qu'on devrait se demander c'est: si valid override la validation native ou les deux doivent fonctionner ensemble?

Reply:
Le props doit toujours avoir precedence, c'est l'interface qui relaye l'intention du user. Si le user specifie que le component est valid, ca sera valid (ou l'inverse), on ne pourra pas faire le contraire.  Dans ton cas, c'est un beau challenge, j'aurais tendance a dire que la solution serait d'enlever les contraintes native momentanement lorsque valid est defini.

Reply:
@savutsang a été plus rapide que moi, mais j'allais dire la même chose:

D'un point de vue DX si on met un prop `valid` on s'attend à avoir le contrôle sur l'apparition du message d'erreur. Donc si on passe un `valid`=`true` on ne s'attend pas à voir éventuellement un message d'erreur à cause de la validation native.

Je comprend ton point de vue, ce que tu suggérais nous éviterait de réécrire les validations natives tel que la contrainte `required`. Mais je pense que si on envoie le prop `valid` on s'attendrait donc que la validation soit faite en amont par nous-même (et les validations natives inclues).

Reply:
En effet, c'est un peu comme les controlled components. Une fois qu'un input est controlled, react impose la value et remplace complètement le comportement 'interne' du input.

Histoire de faciliter la chose, le DS pourrait fournir un hook `useValidation` qui se rattache bien avec nos components validables. On pourrait fournir une petite librairie de contraintes populaires telles `required`, `min`/`max`/`step`, `max-length`, etc. Ou il existe p-e une librairie pour ça, à explorer!

Reply:
Dans le même esprit, l'appel à `setValidity` dans `handleOnInvalid` devrait aussi être conditionnel à `if (valid === undefined)`